platform/linux-generic/odp_init.c

```diff
@@ -173,9 +173,6 @@ static int term_global(enum init_stage stage)
 		}
 		/* Fall through */
 
-	case RANDOM_INIT:
-		/* Fall through */
-
```

Comment:
I am not sure this is a good change as I think it makes the code slightly less robust for future changes. The RANDOM_INIT still exists in the enum.

Reply:
RANDOM_INIT is used in local init/term side.

Local init/term uses only part of the enums. After this change global init/term would also have only those enums that actually do something.

Reply:
Yes, but since it is in the enum and the enum seems to imply certain ordering (even though it is the code that really does), it seems a bit questionable to me to change the behaviour of term_global() if called with RANDOM_INIT. If someone adds a RANDOM_INIT stage in init_global() in the place it is now (implied by the ordering in the enum) and does not need any term handling and does not add the switch case back, we may have a bug.

Or to put it in other way, I do not see how this change makes the code any better, other than maybe making coverity happy. But in that case we could just suppress the warning.

Reply:
Init and term function sequences need to match. If random init global would be added later, it would need to be added into correct position relative to other inits. The term the stage would be added into the matching position. These positions are not dictated by the enum, but which code block uses which other code blocks.

So, I think it's better to remove this dead code, which may be placed wrongly anyway. Similarly, local init/term does not have extra stages.

Reply:
> The term the stage would be added into the matching position.

Only if the term would be needed. There could be init without any associated term.

> These positions are not dictated by the enum, but which code block uses which other code blocks.

Yes, which is why I added: "(even though it is the code that really does)".

> So, I think it's better to remove this dead code, which may be placed wrongly anyway. Similarly, local init/term does not have extra stages.

That is a valid point too. We can leave the code as it is in the patch.